src/plopp/graphics/scatterfig.py

```diff
@@ -0,0 +1,160 @@
+
+from .fig3d import Fig3d
+from ..widgets import Cut3dTool
+from ..widgets.styling import BUTTON_LAYOUT
+
+import scipp as sc
+import ipywidgets as ipw
+
+
+class ScatterFig(Fig3d):
+
+    def __init__(self,
+                 *nodes,
+                 x,
+                 y,
+                 z,
+                 figsize=(600, 400),
+                 title=None,
+                 cbar=None,
+                 **kwargs):
+
+        self._x = x
+        self._y = y
+        self._z = z
+        self._kwargs = kwargs
+        self._figheight = figsize[1]
+
+        super().__init__(*nodes, figsize=figsize, title=title)
+
+        limits = self.get_limits()
+        self.cut_x = Cut3dTool(*nodes,
+                               direction='x',
+                               limits=limits,
+                               view=self,
+                               description='X',
+                               icon='cube')
+        self.cut_y = Cut3dTool(*nodes,
+                               direction='y',
+                               limits=limits,
+                               view=self,
+                               description='Y',
+                               icon='cube')
+        self.cut_z = Cut3dTool(*nodes,
+                               direction='z',
+                               limits=limits,
+                               view=self,
+                               description='Z',
+                               icon='cube')
+        space = ipw.HTML('&nbsp;&nbsp;&nbsp;&nbsp;')
+        self.cutter = ipw.HBox([self.cut_x, space, self.cut_y, space, self.cut_z])
+        self.bottom_bar.add(self.cutter)
+        self.scene.add([self.cut_x.outline, self.cut_y.outline, self.cut_z.outline])
+
+        self.opacity_slider = ipw.FloatSlider(min=0,
+                                              max=0.5,
+                                              step=0.01,
+                                              description='\u03b1',
+                                              orientation='vertical',
+                                              disabled=True,
+                                              value=0.03,
+                                              layout={
+                                                  **BUTTON_LAYOUT['layout'],
+                                                  **{
+                                                      'height': '150px'
+                                                  }
+                                              })
+        self.opacity_slider.observe(self._update_opacity, names='value')
+        self.left_bar.add(self.opacity_slider)
+
+        self._original_children = list(self._children.keys())
+        self.cut_x.button.observe(self._toggle_opacity, names='value')
+        self.cut_y.button.observe(self._toggle_opacity, names='value')
+        self.cut_z.button.observe(self._toggle_opacity, names='value')
```

Comment:
Does `ScatterFig` have to contain all these widgets? Or could it be as simple as possible and only contain the actual scatter plot. And there would be a function that creates the `scene` and sets up all widgets as done here?

It might have to connect widgets as graph nodes to set up communication between them.